### DISCUSSION TITLE
Drop schema cascade?

### DISCUSSION THREAD

**Discussion by seanzechman** [dalibo/ldap2pg#621] [Q&A] [answered]
Not sure if it is a bug or not.  Our situation might be unique.

We create personal schemas on first login with personal account that has been created with ldap2pg.  This is done with a login trigger.

If user leaves the organization then their account is no longer part of LDAP group used by ldap2pg to create/drop accounts.

When ldap2pg runs I see it reassigns objects but it does not do anything with the schema and then cannot cleanup the user appropriately.

2024-04-08 15:08:20,743 CHANGE:  Terminate running sessions for my_user.
2024-04-08 15:08:20,833 CHANGE:  Reassign my_user's objects and purge ACL in postgres.
2024-04-08 15:08:20,834 CHANGE:  Drop my_user.
2024-04-08 15:08:20,835 CRITICAL:  Error while executing SQL query:
2024-04-08 15:08:20,835 CRITICAL:  role "my_user" cannot be dropped because some objects depend on it
2024-04-08 15:08:20,835 CRITICAL:  DETAIL:  1 object in database my_database

my_database=# drop schema my_user;
DROP SCHEMA
my_database=# drop role my_user;
DROP ROLE
my_database=# exit

Is there a way to have drop schema cascade?  I saw notes about it in #171 but that was closed

**Comment by bersace** [comment]
Is `my_database` returned by `databases_query` ?

**Reply by seanzechman** [reply]
Hello - sorry I missed this question.   I do not understand the question but maybe my response will answer it?  

We have yml files that run via cron 1x per day - my_database is hard-coded in the yml file and not determined by a query.  These yml files are initially created and scheduled using ansible templates and inventory definitions.

**Reply by seanzechman** [reply]
@bersace - I do not understand the question.  Problem is if the user owns a schema then ldap2pg cannot drop that user.

**Reply by bersace** [reply, marked as answer]
ldap2pg will reassign the schema to database owner before dropping user. To do this, ldap2pg loops on every databases returned by `databases_query`. If `my_database` is not returned by `databases_query`, ldap2pg won't reassign schemas in this database. Thus Postgres will refuse to drop role.

**Reply by seanzechman** [reply]
I understand now - databases_query is a directive in the yaml syntax.  I added our database to that (was just postgres before) and now it works properly

**Comment by seanzechman** [comment]
However - @bersace - we would want these schemas to be dropped and not just have the ownership changed - is that possible through ldap2pg tool?

**Reply by bersace** [reply]
Hi @seanzechman . ldap2pg does not drop objects. I'm thinking fo a kind of *hook* to be executed before and after creation and drop. See #565.
